l10n_ch_payment_slip/wizard/bvr_import.py

```diff
@@ -198,10 +198,12 @@ def _reconstruct_invoice_ref(self, reference):
                 invoice_move_id = invoice.move_id.id
                 break
         if invoice_move_id:
-            # hack jool: added check with date_maturity
-            move_lines = account_move_line.search([('move_id', '=', invoice_move_id),
-                                                   ('reconciled', '=', False)])
-            ret = move_lines
+            move_line = account_move_line.search([('move_id', '=', invoice_move_id),
+                                                  ('reconciled', '=', False),
+                                                  ('account_id.user_type_id.type', 'in', ['receivable', 'payable']),
+                                                  ('journal_id.type', '=', 'sale')],
+                                                 limit=1)
```

Comment:
We can either use limit=1
or
use something like:
https://github.com/brain-tec/l10n-switzerland/blob/9.0/l10n_ch_payment_slip/wizard/bvr_import.py#L188